static/js/stream_list.js

```diff
@@ -145,13 +145,16 @@ export function build_stream_list(force_rerender) {
     const any_pinned_streams = stream_groups.pinned_streams.length > 0;
     const any_normal_streams = stream_groups.normal_streams.length > 0;
     const any_dormant_streams = stream_groups.dormant_streams.length > 0;
+    const any_muted_pinned_streams = stream_groups.muted_pinned_streams.length > 0;
+    const any_muted_normal_streams = stream_groups.muted_active_streams.length > 0;
+
     const need_section_subheaders =
-        (any_pinned_streams ? 1 : 0) +
-            (any_normal_streams ? 1 : 0) +
+        (any_normal_streams || any_muted_normal_streams ? 1 : 0) +
+            (any_pinned_streams || any_muted_pinned_streams ? 1 : 0) +
```

Comment:
I think a better fix is actually that we should define `any_pinned_streams = stream_groups.pinned_streams.length > 0 || stream_groups.muted_pinned_streams.length > 0;`; rather than repeating these combinations in like 5 places.

Reply:
Roger that!